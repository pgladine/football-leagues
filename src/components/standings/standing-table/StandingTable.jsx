import React, { useEffect, useState } from 'react';
import { useParams } from 'react-router-dom';
import standingsService from '../../../services/standings';
import StandingRow from '../standing-row/StandingRow'
import './standing-table.css'
import Loader from '../../loader/Loader';

export default function StandingTable() {
  const [standing, setStanding] = useState(null)
  const { leagueId } = useParams()

  useEffect(() => {
    standingsService.list(leagueId)
      .then(setStanding)
      .catch(console.error)
  }, [leagueId]);

  return (
    <section>
      {!standing ? (<Loader />) : (
          <>
          {standing.map((leagueGroup, i) => (
            <div className='standing-table-container' key={i}>
              {leagueGroup[0] && <p>{leagueGroup[0]}</p>}
              <div className='standing-header' data-testid='standing-list-header'>
                <p>EQUIPOS</p>
                <div className='team-numbers'>
                  <span className='total-points'>PT</span>
                  <span>PJ</span>
                  <span>PG</span>
                  <span>PE</span>
                  <span>PP</span>
                  <span>GF</span>
                  <span>GC</span>
                </div>
              </div>
              {leagueGroup[1].map((teamData) => (
                <StandingRow key={teamData.teamId} {...teamData} />
              ))}
            </div>
          ))}
          </>
        )
      }
    </section>
  )
}
