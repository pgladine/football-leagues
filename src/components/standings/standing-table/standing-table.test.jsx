import { render, screen, cleanup, waitFor } from "@testing-library/react";
import { vi } from 'vitest'
import StandingTable from "./StandingTable";
import standingsService from '../../../services/standings'

describe('<StandingTable/>', () => {
  afterEach(cleanup)

  it('should render <Loader/> when standing is null ', () => {
    // Given
    const standingsServiceListMock = vi.spyOn(standingsService, 'list').mockResolvedValue(null)


    // When
    render(<StandingTable />)

    // Then
    expect(standingsServiceListMock).toHaveBeenCalledTimes(1);
    expect(screen.getByTestId('loader')).toBeInTheDocument();
    expect(document.body).toMatchSnapshot();
  })

  it('should render a list of <StandingRow/> component with header', async () => {
    // Given
    const mockStandingLeagueList = [
      [null, [
        {
          position: 1,
          teamId: 3,
          teamCrest: 'https://teamCrest1.url',
          teamName: 'teamName1',
          points: 15,
          playedGames: 6,
          won: 5,
          draw: 0,
          lost: 1,
          goalsFor: 20,
          goalsAgainst: 6,
        },
        {
          position: 2,
          teamId: 4,
          teamCrest: 'https://teamCrest2.url',
          teamName: 'teamName2',
          points: 14,
          playedGames: 6,
          won: 4,
          draw: 0,
          lost: 2,
          goalsFor: 18,
          goalsAgainst: 8,
        }
      ]]
    ]
    const standingsServiceListMock = vi.spyOn(standingsService, 'list').mockResolvedValue(mockStandingLeagueList)

    // When
    render(<StandingTable />);

    // Then
    await waitFor(async () =>
      expect(await screen.findByTestId('loader')).not.toBeInTheDocument()
    )

    expect(standingsServiceListMock).toHaveBeenCalledTimes(1);
    expect(await screen.findByTestId('standing-list-header')).toBeInTheDocument();
    // TODO ver que pinta 2 filas
    expect((await screen.findAllByTestId('standing-row')).length).toEqual(2)
  })
})