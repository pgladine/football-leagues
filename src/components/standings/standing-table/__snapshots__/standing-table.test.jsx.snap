// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<StandingTable/> > should render <Loader/> when standing is null  1`] = `
<body>
  <div>
    <section>
      <div
        class="box"
        data-testid="loader"
      >
        <div
          class="shadow"
        />
        <div
          class="gravity"
        >
          <div
            class="ball"
          />
        </div>
      </div>
    </section>
  </div>
</body>
`;
